Sources/ProjectSpec/Target.swift

```diff
@@ -159,6 +162,63 @@ extension Target {
         merged["targets"] = crossPlatformTargets
         return merged
     }
+    
+    public var shouldEmbedDependencies: Bool {
+        return type.isApp || type.isTest
+    }
+    
+    public func generateEffectiveDependencies(project: Project) -> [Dependency] {
+        if transitivelyLinkDependencies ?? project.options.transitivelyLinkDependencies {
+            return getAllDependenciesPlusTransitiveNeedingEmbedding(project: project)
+        } else {
+            return dependencies
+        }
+    }
+    
+    private func getAllDependenciesPlusTransitiveNeedingEmbedding(project: Project) -> [Dependency] {
```

Comment:
I'd rather keep all this code out of the `ProjectSpec` module. Some people use `ProjectSpec` as a seperate library without XcodeGen's logic.
But mostly this also removes the ability to add caching across the project as there is no controller for this logic

Reply:
Removed that commit altogether, it's back in `XcodeGenKit`.